### ISSUE TITLE
How does it work?

### ISSUE THREAD

**Issue by domluna** [chriselrod/JuliaFormatterDaemonMode.jl#2]
Looks cool, is that you have a running process you can call to format a directory/file?

not an issue 🙌

**Comment by chriselrod** [comment]
I haven't been using it, but I added an example to the README:
```sh
$ julia --project=~/.julia/dev/JuliaFormatterDaemonMode/ -e 'using Pkg; Pkg.update()'
$ julia --project=~/.julia/dev/JuliaFormatterDaemonMode/ -e 'using DaemonMode; serve()'&
$ time ~/.julia/dev/JuliaFormatterDaemonMode/bin/juliafmt ~/.julia/dev/Octavian/
2.305 secs
$ time ~/.julia/dev/JuliaFormatterDaemonMode/bin/juliafmt ~/.julia/dev/Octavian/
0.334 secs
```

Basically, start a server, and then run `bin/juliafmt` to format.
In my experiments, this was significantly faster than `PackageCompiler.jl` compiled binaries.
A better precompile script probably would help, but it also takes a really long time to produce. I think DaemonMode.jl is a much more practical solution to Julia scripting than PackageCompiler.jl.

I use emacs, so I think I'll soon switch to
https://codeberg.org/FelipeLema/julia-formatter.el
which also manages a server, but will start it up automatically.
Currently, I'm using the langauge server, which is of course a server, but it starts another one up for each Julia project, and `julia-formatter.el` apparently takes care of making it asynchronous, and my lazy code that automatically calls it when I save is blocking (which is why I kept looking for hacks to improve the performance)...

If I had more Julia scripts, I'd probably embrace `DaemonMode.jl` more fully, make sure I always have a sever running, and then use JuliaFormatterDaemonMode.jl.

FWIW, I ultimately decided that rather than spending any more time looking at using things like PackageCompiler, I'd rather just try to improve the latency by looking for type instabilities, invalidations, etc. But I haven't really had the time for that either.

**Comment by domluna** [comment]
makes sense https://github.com/domluna/JuliaFormatter.jl/pull/685 recently fixed some type instabilities with JET. In my timings it speeds up the first format by about 30% and the followup by a little less, 10-15%.

I'm going to look into fixing the rest of the instabilities mentioned in the PR

**Comment by chriselrod** [comment]
Nice! Yeah, much better to treat the disease (bad load times) than the symptoms (hide them via async and package compilation).

**Comment by domluna** [comment]
https://github.com/domluna/JuliaFormatter.jl/pull/689
